controllers/velero_test.go

```diff
@@ -588,6 +588,117 @@ func TestDPAReconciler_buildVeleroDeployment(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "given valid DPA CR with Unsupported Server Args, appropriate Velero Deployment is built",
+			veleroDeployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-velero-deployment",
+					Namespace: "test-ns",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{MatchLabels: veleroDeploymentMatchLabels},
+				},
+			},
+			dpa: &oadpv1alpha1.DataProtectionApplication{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-Velero-CR",
+					Namespace: "test-ns",
+					Annotations: map[string]string{
+						"oadp.openshift.io/unsupported-velero-server-args": "unsupported-server-args-cm",
+					},
+				},
+				Spec: oadpv1alpha1.DataProtectionApplicationSpec{
+					Configuration: &oadpv1alpha1.ApplicationConfig{
+						Velero: &oadpv1alpha1.VeleroConfig{},
```

Comment:
would add a new test, or do in this one, to confirm that any value user adds here is overwritten

Reply:
In the last commit added above test scenario and as well scenario where the ConfigMap name in the annotation was empty. I believe we now have really nice set of different tests for this feature.